### PR TITLE
Allow vehicle tables to perform butchery

### LINKED ISSUE #29010: check vehicle parts for butchering surface
As was pointed out in #28511 the butchery code doesn't check for flat surfaces installed in vehicles. It does check for kitchens, but that's a bit overboard sometimes.

**Describe the solution you'd like**  
https://github.com/CleverRaven/Cataclysm-DDA/blob/71fd5f239e0ede4ac2703080b09c619ed995cfd6/src/activity_handlers.cpp#L322-L328
Looks like you can add something like ``g->m.veh_at( pt )->has_feature( "FLAT_SURF" )`` to the above code and add "FLAT_SURF" to
https://github.com/CleverRaven/Cataclysm-DDA/blob/b42b941b9b25d25128d2b087f35af5ee1bb511a8/data/json/vehicle_parts.json#L978-L998
and it might do what you want.

**Describe alternatives you've considered**
Not allowing butchery on vehicle flat surfaces, I cant see a compelling reason though.

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Bugfixes "Allow vehicle tables to perform butchery"```

#### Purpose of change

FIxes #29010 

#### Describe the solution
added FLAT_SURF vehicle part json flag.
checked for that vehicle flag in activity_handlers.cpp

#### Describe alternatives you've considered
N/A

#### Additional context
N/A

### CODE REVIEW

**Comment by Reclusive-reptile** [comment]
``"veh_table_wood"`` Should also get the flag.

**Comment by davidpwbrown** [comment]
> `"veh_table_wood"` Should also get the flag.

Ah yeah thanks, forgot about that one

File: src/veh_type.h [hunk 1]
```diff
@@ -80,6 +80,7 @@ enum vpart_bitflags : int {
     VPFLAG_WASHING_MACHINE,
     VPFLAG_FLUIDTANK,
     VPFLAG_REACTOR,
+    VPFLAG_FLAT_SURF,
```
Comment (mlangsdorf) [inline src/veh_type.h]: quoting the header file:
```
// bitmask backing store of -certain- vpart_info.flags, ones that
// won't be going away, are involved in core functionality, and are checked frequently
enum vpart_bitflags : int {
```
this is not core functionality nor is it checked frequently.  do not put it in the vpart_bitflags.

File: src/veh_type.cpp [hunk 2]
```diff
@@ -87,6 +87,7 @@ static const std::unordered_map<std::string, vpart_bitflags> vpart_bitflag_map =
     { "WASHING_MACHINE", VPFLAG_WASHING_MACHINE },
     { "FLUIDTANK", VPFLAG_FLUIDTANK },
     { "REACTOR", VPFLAG_REACTOR },
+    { "FLAT_SURF", VPFLAG_FLAT_SURF },
```
Comment (mlangsdorf) [inline src/veh_type.cpp]: See below, but no, don't do this.

**Pull request merged**